src/lib.rs

```diff
@@ -4442,6 +4443,90 @@ pub fn digraph_spiral_layout(
     layout::spiral_layout(&graph.graph, scale, center, resolution, equidistant)
 }
 
+fn _num_shortest_paths_unweighted<Ty: EdgeType>(
+    graph: &StableGraph<PyObject, PyObject, Ty>,
+    source: usize,
+) -> PyResult<HashMap<usize, u128>> {
+    let mut out_map: Vec<Option<u128>> = vec![None; graph.node_bound()];
+    let node_index = NodeIndex::new(source);
+    if graph.node_weight(node_index).is_none() {
+        return Err(PyIndexError::new_err(format!(
+            "No node found for index {}",
+            source
+        )));
+    }
+    let mut bfs = Bfs::new(&graph, node_index);
+    let mut distance: Vec<Option<u128>> = vec![None; graph.node_bound()];
```

Comment:
`distance` does not need to be `u128`, `usize` is sufficient. The distance between two nodes is bounded by the number of nodes in a graph, which fits in a `usize` value.`out_map` is the only result that needs to be calculated as `u128/BigInt`.

Reply:
Done in https://github.com/Qiskit/retworkx/pull/355/commits/abbc61fddc881b8afe69f70af2794fde3742432e